### PR TITLE
commands/.../test/local,pkg/test: add image flag to test local

### LINKED ISSUE #746: Add flag to allow a user to specify their own image for local tests
## Feature Request

**Is your feature request related to a problem? Please describe.**
Currently, if a user wants to use a different image for their operator when running local tests, they need to manually modify the image section of their `operator.yaml` file to use the correct image. In automated CI systems, it would be nice to simply specify the image name with a flag.

**Describe the solution you'd like**
We could add a flag that updated the `image` section of the deployment to the image the user specified.

cc @aravindhp

### PR DESCRIPTION
**Pull request by AlexNPavel**
**Description of the change:** Adds a `--image` flag to the `test local` subcommand. Currently only supports 1 deployment with 1 container (we may add support for multi-deployment, multi-container manifests in the future, but it would also require more flags from the user to identify which image needs to be replaced).

**Motivation for the change:** See #746 

Resolves #746 

cc @awgreene

### CODE REVIEW

File: commands/operator-sdk/cmd/test/local.go [hunk 1]
```diff
@@ -176,3 +189,69 @@ func combineManifests(base, manifest []byte) []byte {
...
+		if string(yamlSpec) == "" {
+			continue
+		}
+		decoded := make(map[string]interface{})
+		err = yaml.Unmarshal(yamlSpec, &decoded)
+		if err != nil {
+			return err
+		}
+		kind, ok := decoded["kind"].(string)
+		if !ok {
```
Comment (estroz) [inline commands/operator-sdk/cmd/test/local.go]: This and the `kind != "Deployment"` branch can be combined since they're doing the same thing.

File: commands/operator-sdk/cmd/test/local.go [hunk 2]
```diff
@@ -141,6 +149,11 @@ func testLocalFunc(cmd *cobra.Command, args []string) {
 			}
 		}()
 	}
+	if tlConfig.image != "" {
+		if err := replaceImage(tlConfig.namespacedManPath, tlConfig.image); err != nil {
```
Comment (estroz) [inline commands/operator-sdk/cmd/test/local.go]: Any reason we can't replace the image before combining the manifests, i.e. [here](https://github.com/operator-framework/operator-sdk/blob/master/commands/operator-sdk/cmd/test/local.go#L88-L91)?

Comment (AlexNPavel) [reply commands/operator-sdk/cmd/test/local.go]: We can't do that because a user can pass in their own namespaced manifest, in which case we would skip the combination step.

File: commands/operator-sdk/cmd/test/local.go [hunk 3]
```diff
@@ -141,6 +149,11 @@ func testLocalFunc(cmd *cobra.Command, args []string) {
...
+			log.Fatalf("replaceImage function failed: %v", err)
```
Comment (hasbro17) [inline commands/operator-sdk/cmd/test/local.go]: Probably rephrase this to be better understood by the user:
```Go
log.Fatalf("failed to overwrite operator image in the namespaced manifest: %v", err)
```

Comment (AlexNPavel) [reply commands/operator-sdk/cmd/test/local.go]: Yup, forgot to reword that after debugging.

File: CHANGELOG.md [hunk 4]
```diff
@@ -3,6 +3,7 @@
 ### Changed
 
 - The SDK now uses logr as the default logger to unify the logging output with the controller-runtime logs. Users can still use a logger of their own choice. See the [logging doc](https://github.com/operator-framework/operator-sdk/blob/master/doc/user/logging.md) on how the SDK initializes and uses logr.
+- Add `image` flag to `test local` subcommand ([#768](https://github.com/operator-framework/operator-sdk/pull/768))
```
Comment (hasbro17) [inline CHANGELOG.md]: Need to update the CLI reference flags for `operator-sdk test local` to include the `--image` flag and hyperlink to that here.
https://github.com/operator-framework/operator-sdk/blob/master/doc/sdk-cli-reference.md#local

**Comment by hasbro17** [comment]
@AlexNPavel Can we add a one line explanation and example usage of this flag in the [writing-e2e-tests doc](https://github.com/operator-framework/operator-sdk/blob/master/doc/test-framework/writing-e2e-tests.md#local)?

**Comment by AlexNPavel** [comment]
@hasbro17 Added documentation

File: doc/sdk-cli-reference.md [hunk 5]
```diff
@@ -257,6 +257,7 @@ Runs the tests locally
 * `--namespaced-manifest` string - path to manifest for per-test, namespaced resources (default: combines deploy/service_account.yaml, deploy/rbac.yaml, and deploy/operator.yaml)
 *  `--namespace` string - if non-empty, single namespace to run tests in (e.g. "operator-test") (default: "")
 * `--go-test-flags` string - extra arguments to pass to `go test` (e.g. -f "-v -parallel=2")
+* `--image` string - Use a different image from the one specified in the namespaced manifest
```
Comment (hasbro17) [inline doc/sdk-cli-reference.md]: `different image` ==> `different operator image`
